# SUBLYME: Software for Uncovering Bacteriophage LYsins in MEtagenomic datasets using protein embeddings


## Installation & usage

First create a virtual environment, then: 

**From pypi**:
```
pip install numpy, pandas, transformers, sublyme

python
from sublyme import sublyme
```

ex. `sublyme()`


**From source**:
```
git clone https://github.com/Rousseau-Team/sublyme.git

pip install numpy pandas transformers
```

ex. `python sublyme/src/sublyme/sublyme.py -h`

## Advanced usage
**Positional arguments**:
- **input_file** - Input file


**Optional arguments**:
- **output_file** - File to save table of segmented domains in csv format. (default spaed_predictions.csv)
- **models_folder** - Path to fasta file or folder containing fasta files. If specified, spaed will save the sequences corresponding to predicted domains into a new fasta file named  "spaed_predicted_domains.faa" in the same output folder as output_file. Ensure fasta names or headers correspond to entries in pae files.
- **only_embeddings** - Maximum number of clusters initially generated by hierarchical clustering corresponds to len(protein) // RATIO_NUM_CLUSTERS. (Default 10). For a protein 400 residues long, 40 clusters will be generated.
